File: apps/api/app/schemas/agent.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.agent import AgentStatus


class AgentRead(BaseModel):
    id: uuid.UUID
    name: str
    slug: str
    description: str | None
    status: AgentStatus
    capabilities: list[str] | None
    supported_task_types: list[str] | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AgentList(BaseModel):
    items: list[AgentRead]
    total: int

File: apps/web/types/agent.ts
/** Agent types matching the backend schemas. */

export type AgentStatus = "active" | "inactive" | "deprecated";

export interface Agent {
  id: string;
  name: string;
  slug: string;
  description: string | null;
  status: AgentStatus;
  capabilities: string[] | null;
  supported_task_types: string[] | null;
  created_at: string;
  updated_at: string;
}

export interface AgentList {
  items: Agent[];
  total: number;
}
